test/test_data/emacs.py

```diff
@@ -0,0 +1,40 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+mapping_test_data = [
+    {
+        "product": "emacs",
+        "version": "28.2",
+        "version_strings": [
+            r"$Id: GNU Emacs 28.2"
+        ],
+    },
+    {
+        "product": "emacs",
+        "version": "25.2",
+        "version_strings": [
+            r"$Id: GNU Emacs 25.2"
+        ],
+    },
+    {
+        "product": "emacs",
+        "version": "26.1",
+        "version_strings": [
+            r"$Id: GNU Emacs 26.1"
+        ],
+    },
+]
+package_test_data = [
+    {
+        "url": "http://ports.ubuntu.com/pool/main/e/emacs25/",
+        "package_name": "emacs25_25.2+1-6_arm64.deb",
+        "product": "emacs",
+        "version": "25.2",
```

Comment:
You're getting the following error:

```python
FAILED test/test_scanner.py::TestScanner::test_version_in_package[http://ports.ubuntu.com/pool/main/e/emacs25/-emacs25_25.2+1-6_arm64.deb-emacs-25.2-other_products167] - AssertionError: libtiff found in emacs25_25.2+1-6_arm64.deb. If that's expected, make sure to add libtiff to the expected list of other_products.
assert 'libtiff' not in {'emacs', 'libtiff', 'pango'}
```

Assuming libtiff (and pango) are in fact part of the ubuntu emacs package then you just need to declare them as expected using the "other_products" field as so:

```suggestion
        "version": "25.2",
        "other_products": ["libtiff", "pango"],
```

If they're *not* expected and are actually false positives, we might need to adjust the signatures in libtiff and/or pango so that they don't match your binaries. That could be a separate issue, though (e.g. it's fine to fix this one with other_products and open a new issue for "fix the libtiff signature so it doens't match the emacs test")

Reply:
We've been fine tuning these as we go -- so far it's mostly been doable to add more to signatures to avoid these (for example, often the signature can add in the next or preivous line, which won't be the same in the original binary as in where it's shared).  But yes, let's get an issue open for at least libtiff and pango to see if we can improve those for the future.